AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -94,6 +97,16 @@ - (void)setDelegate:(id<ASDataControllerDelegate>)delegate
   _delegateDidDeleteSections  = [_delegate respondsToSelector:@selector(dataController:didDeleteSectionsAtIndexSet:withAnimationOptions:)];
 }
 
+- (void)setDataSource:(id<ASDataControllerSource>)dataSource {
+  if (_dataSource == dataSource) {
+    return;
+  }
+
+  _dataSource = dataSource;
+  // This probably won't be sufficient to tell if we should call the node block
```

Comment:
Why would it not be sufficient? It seems to me that if we implement the node block version, we would not ever care to call the other version